shakedown/cli/main.py

```diff
@@ -78,68 +78,92 @@ def cli(**args):
 
         echo(getattr(imported[req], requirements[req]))
 
-    if args['ssl_no_verify']:
-        imported['dcos'].config.set_val('core.ssl_verify', 'False')
-
     echo('Checking for DC/OS cluster...', d='step-min', n=False)
 
-    with stdchannel_redirected(sys.stderr, os.devnull):
-        imported['dcos'].config.set_val('core.dcos_url', args['dcos_url'])
+    attached = False
 
     try:
-        echo(shakedown.dcos_version())
-    except:
-        click.secho("error: cluster '" + args['dcos_url'] + "' is unreachable.", fg='red', bold=True)
-        sys.exit(1)
-
-    echo('Authenticating with cluster...', d='step-maj')
-    authenticated = False
-    token = imported['dcos'].config.get_config_val("core.dcos_acs_token")
-    if token is not None:
-        echo('Validating existing ACS token...', d='step-min', n=False)
-        try:
-            shakedown.dcos_leader()
-
-            echo('ok')
-            authenticated = True
-        except imported['dcos'].errors.DCOSException:
-            click.secho("error: authentication failed.", fg='red', bold=True)
-    if not authenticated and args['oauth_token']:
-       try:
-            echo('Validating OAuth token...', d='step-min', n=False)
-            token = shakedown.authenticate_oauth(args['oauth_token'])
-
-            with stdchannel_redirected(sys.stderr, os.devnull):
-                imported['dcos'].config.set_val('core.dcos_acs_token', token)
-
-            authenticated = True
-            echo('ok')
-       except:
-            click.secho("error: authentication failed.", fg='red', bold=True)
-    if not authenticated and args['username'] and args['password']:
-        try:
-            echo('Validating username and password...', d='step-min', n=False)
-            token = shakedown.authenticate(args['username'], args['password'])
-
-            with stdchannel_redirected(sys.stderr, os.devnull):
-                imported['dcos'].config.set_val('core.dcos_acs_token', token)
+        if args['dcos_url'] == imported['dcos'].config.get_config_val('core.dcos_url'):
+            try:
```

Comment:
Yes, they're trying different scenarios that can fail for different reasons,

- outer: does `imported['dcos'].config.get_config_val()` throw an exception (it will if a cluster isn't attached)
- inner: if a cluster is attached and it's URL matches `args['dcos_url']`, can a cluster version be pulled?  if unauthenticated, another `DCOSException` is thrown here.